File: codeforces/low_difficulty/difficulty_800/78.bachgold_problem_[749A]/bachgold_problem.py
# Author: José Rodolfo (jric2002)
n = int(input())
if ((n % 2) == 0):
  amount_two = int(n / 2)
  print(amount_two)
  for i in range(0, amount_two):
    print(2, end=" ")
  print(end="\n")
else:
  amount_two = int((n - 3) / 2)
  print(amount_two + 1)
  for i in range(0, amount_two):
    print(2, end=" ")
  print(3, end="\n")

File: codeforces/low_difficulty/difficulty_800/78.bachgold_problem_[749A]/bachgold_problem.cpp
#include <iostream>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned int n, amount_two;
  cin >> n;
  if ((n % 2) == 0) {
    amount_two = n / 2;
    cout << amount_two << endl;
    for (unsigned int i = 0; i < amount_two; i++) {
      cout << 2 << " ";
    }
    cout << endl;
  }
  else {
    amount_two = (n - 3) / 2;
    cout << (amount_two + 1) << endl;
    for (unsigned int i = 0; i < amount_two; i++) {
      cout << 2 << " ";
    }
    cout << 3 << endl;
  }
  return 0;
}